content/en/blog/_posts/2023-08-04-kubernetes-1.28-blog.md

```diff
@@ -0,0 +1,275 @@
+    
+CRDs need direct support for non-trivial validation. While admission webhooks do support CRDs validation, they significantly complicate the development and operability of CRDs.
+
+This KEP proposes that an inline expression language be integrated directly into CRDs such that a much larger portion of validation use cases can be solved without the use of webhooks. When selecting an expression language, we want to be sure that it can support defaulting and CRD conversion in the future.
+
+It is sufficiently lightweight and safe to be run directly in the kube-apiserver (since CRD creation is a privileged operation), has a straight-forward and unsurprising grammar, and supports pre-parsing and type checking of expressions, allowing syntax and type errors to be caught at CRD registration time.
+
+## ValidatingAdmissionPolicies graduate to beta
+    
+Common Expression language for admission control is customizable, in-process validation of requests to the Kubernetes API server as an alternative to validating admission webhooks.
+
+This builds on the capabilities of the CRD Validation Rules feature that graduated to beta in 1.25 but with a focus on the policy enforcement capabilities of validating admission control.
+    
+This will lower the infrastructure barrier to enforcing customizable policies as well as providing primitives that help the community establish and adhere to the best practices of both K8s and its extensions.
+
+## CRD validation ratcheting
+    
+This Allows custom resources to fail validation if the patch did not alter any of the invalid fields.  The ability to shift left validation logic from controllers to the front end is a long-term goal for improving the useability of the Kubernetes project.
+  
+## Match conditions for admission webhooks
+    
+Kubernetes v1.27 lets you specify _match conditions_ for admission webhooks,
+which lets you narrow the scope of when Kubernetes makes a remote HTTP call at admission time.
+The `matchCondition` field for ValidatingWebhookConfiguration and MutatingWebhookConfiguration
+is a CEL expression that must evaluate to true for the admission request to be sent to the webhook.
+
+In Kubernetes v1.28, that field moved to beta, and it's enabled by default.
+    
+To learn more, see [`matchConditions`](/docs/reference/access-authn-authz/extensible-admission-controllers/#matching-requests-matchconditions) in the Kubernetes documentation.
+
+## Changes to supported skew between control plane and node versions
+    
+This enables testing and expanding the supported skew between core node and control plane components by one version from n-2 to n-3, so that node components (kubelet and kube-proxy) for the oldest supported minor version work with control plane components (kube-apiserver, kube-scheduler, kube-controller-manager, cloud-controller-manager) for the newest supported minor version.
```

Comment:
I think this is important for end users. It essentially means they can upgrade the control plane a little faster than their nodes (which are almost always going to be the longer thing with running workloads). 

If I was on 1.25 and I wanted to go to 1.28 previously, I would need to upgrade my control plane to 1.26, then 1.27, then I would need to upgrade my nodes to 1.26, then I could upgrade my control plane to 1.28. This will allow upgrades of the control plan 1.25 -> 1.26 -> 1.27 -> 1.28 and then I could start upgrading my nodes.

Reply:
TL;DR I think this makes it better align with the yearly support period and reduces operational burden on kubernetes operators.